experimental/oak_async/src/io.rs

```diff
@@ -72,6 +73,32 @@ impl<T: Decodable> Drop for ChannelRead<T> {
     }
 }
 
+/// `Stream` representing a sequence of asynchronous reads from a channel.
+pub struct ChannelReadStream<T: Decodable>(
+    // Note: `Stream` could be implemented directly on the `ChannelRead` type. Unfortunately the
+    // `Future` and `Stream` extension traits have some methods that overlap, such as `map`. This
```

Comment:
Why not just directly impl `Stream` for `ChannelRead`, and forget about `Future`?

Reply:
I expect that reading a single message from a channel (and expecting one to be sent) will be a fairly common pattern that I think is a bit clumsy if we just have a `Stream`. There are things like `next` and `into_future`, but neither of these are convenient for this purpose.

When we make async the default I think it would be reasonable to make `Receiver` implement `Stream` directly, keeping `receive_async` around to have a convenient method for a `Future`. I am wondering if there could ever be a case where 'conflating' a receiver and `Stream` into one type could lead to issues, let me know if you can think of anything. It seems like they could become one and the same.

Having separate types might also help with type safety, especially since we have methods in gRPC that can be streaming or non-streaming. Having two different types could help express that better.